shell/platform/linux/fl_keyboard_manager.h

```diff
@@ -36,68 +22,40 @@ G_DECLARE_FINAL_TYPE(FlKeyboardManager,
 /**
  * FlKeyboardManager:
  *
- * A hub that manages how key events are dispatched to various processing
- * objects of Flutter, or possibly back to the system.
- *
- * This class manage one or more objects of #FlKeyResponder, as well as a
- * #TextInputPlugin.
- *
- * An event that is received by #fl_keyboard_manager_handle_event is first
- * dispatched to *all* responders. Each responder responds *ascynchronously*
- * with a boolean, indicating whether it handles the event.
+ * Processes keyboard events and cooperate with `TextInputPlugin`.
  *
- * An event that is not handled by any responders is then passed to to the
- * #TextInputPlugin, which responds *synchronously* with a boolean, indicating
- * whether it handles the event.
+ * A keyboard event goes through a few sections, each can choose to handled the
+ * event, and only unhandled events can move to the next section:
  *
- * If no processing objects handle the event, the event is then "redispatched":
- * sent back to the system using #redispatch_callback.
- *
- * Preventing responders from receiving events is not supported, because in
- * reality this class will only support 2 hardcoded ones (channel and
- * embedder), where the only purpose of supporting two is to support the legacy
- * API (channel) during the deprecation window, after which the channel
- * responder should be removed.
+ * - Keyboard: Dispatch to the embedder responder and the channel responder
+ *   simultaneously. After both responders have responded (asynchronously), the
+ *   event is considered handled if either responder handles.
+ * - Text input: Events are sent to IM filter (usually owned by
+ *   `TextInputPlugin`) and are handled synchronously.
+ * - Redispatching: Events are inserted back to the system for redispatching.
  */
 
 /**
  * fl_keyboard_manager_new:
- * @text_input_plugin: the #FlTextInputPlugin to send key events to if the
- * framework doesn't handle them. This object will be managed and freed by
- * #FlKeyboardManager.
- * @redispatch_callback: how the events should be sent if no processing
- * objects handle the event. Typically a function that calls #gdk_event_put
- * on #FlKeyEvent::origin.
+ * @view_delegate: An interfaces that the manager requires to communicate with
+ * the system. Usually implemented by FlView.
```

Comment:
```suggestion
 * the platform. Usually implemented by FlView.
```
At least I assume that's which system you mean... (system works too, it's just vague). You could also say "Linux" or "GTK" instead of "system", to be more specific.